hedera-node/src/main/java/com/hedera/services/stream/RecordStreamFileWriter.java

```diff
@@ -0,0 +1,490 @@
+package com.hedera.services.stream;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.CodedOutputStream;
+import com.hedera.services.stream.proto.HashAlgorithm;
+import com.hedera.services.stream.proto.HashObject;
+import com.hedera.services.stream.proto.RecordStreamFile;
+import com.hedera.services.stream.proto.RecordStreamItem;
+import com.hedera.services.stream.proto.SignatureFile;
+import com.hedera.services.stream.proto.SignatureObject;
+import com.hedera.services.stream.proto.SignatureType;
+import com.hederahashgraph.api.proto.java.SemanticVersion;
+import com.swirlds.common.crypto.DigestType;
+import com.swirlds.common.crypto.Hash;
+import com.swirlds.common.crypto.HashingOutputStream;
+import com.swirlds.common.crypto.RunningHash;
+import com.swirlds.common.io.streams.SerializableDataOutputStream;
+import com.swirlds.common.stream.LinkedObjectStream;
+import com.swirlds.common.stream.Signer;
+import com.swirlds.logging.LogMarker;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.Instant;
+
+import static com.swirlds.common.stream.LinkedObjectStreamUtilities.generateSigFilePath;
+import static com.swirlds.common.stream.LinkedObjectStreamUtilities.generateStreamFileNameFromInstant;
+import static com.swirlds.common.stream.LinkedObjectStreamUtilities.getPeriod;
+import static com.swirlds.logging.LogMarker.EXCEPTION;
+import static com.swirlds.logging.LogMarker.OBJECT_STREAM;
+import static com.swirlds.logging.LogMarker.OBJECT_STREAM_FILE;
+
+class RecordStreamFileWriter<T extends RecordStreamObject> implements LinkedObjectStream<T> {
```

Comment:
It might be clearer to simply use,
```
class RecordStreamFileWriter implements LinkedObjectStream<RecordStreamObject> {
```
...since `RecordStreamObject` is not particularly designed for inheritance, nor likely to be subclassed.

Reply:
[Fixed](https://github.com/hashgraph/hedera-services/pull/3461/commits/658e777c73e0f12d25d4b3fabf2dd8a102c3b5c6)